File: web/src/api/expertProfile.js
import service from '@/utils/request'

// @Tags ExpertProfile
// @Summary 创建专家主档
// @Router /expertProfile/createExpertProfile [post]
export const createExpertProfile = (data) => {
  return service({
    url: '/expertProfile/createExpertProfile',
    method: 'post',
    data
  })
}

// @Tags ExpertProfile
// @Summary 删除专家主档
// @Router /expertProfile/deleteExpertProfile [delete]
export const deleteExpertProfile = (params) => {
  return service({
    url: '/expertProfile/deleteExpertProfile',
    method: 'delete',
    params
  })
}

// @Tags ExpertProfile
// @Summary 批量删除专家主档
// @Router /expertProfile/deleteExpertProfileByIds [delete]
export const deleteExpertProfileByIds = (params) => {
  return service({
    url: '/expertProfile/deleteExpertProfileByIds',
    method: 'delete',
    params
  })
}

// @Tags ExpertProfile
// @Summary 更新专家主档
// @Router /expertProfile/updateExpertProfile [put]
export const updateExpertProfile = (data) => {
  return service({
    url: '/expertProfile/updateExpertProfile',
    method: 'put',
    data
  })
}

// @Tags ExpertProfile
// @Summary 用id查询专家主档
// @Router /expertProfile/findExpertProfile [get]
export const findExpertProfile = (params) => {
  return service({
    url: '/expertProfile/findExpertProfile',
    method: 'get',
    params
  })
}

// @Tags ExpertProfile
// @Summary 分页获取专家主档列表
// @Router /expertProfile/getExpertProfileList [get]
export const getExpertProfileList = (params) => {
  return service({
    url: '/expertProfile/getExpertProfileList',
    method: 'get',
    params
  })
}

File: server/router/ExpertDatabase/expert_profile.go
package ExpertDatabase

import (
	v1 "github.com/flipped-aurora/gin-vue-admin/server/api/v1"
	"github.com/flipped-aurora/gin-vue-admin/server/middleware"
	"github.com/gin-gonic/gin"
)

type ExpertProfileRouter struct{}

// InitExpertProfileRouter 初始化 专家主档 路由信息
func (s *ExpertProfileRouter) InitExpertProfileRouter(Router *gin.RouterGroup) {
	expertProfileRouter := Router.Group("expertProfile").Use(middleware.OperationRecord())
	expertProfileRouterWithoutRecord := Router.Group("expertProfile")
	expertProfileApi := v1.ApiGroupApp.ExpertDatabaseApiGroup.ExpertProfileApi
	{
		expertProfileRouter.POST("createExpertProfile", expertProfileApi.CreateExpertProfile)             // 新建专家主档
		expertProfileRouter.DELETE("deleteExpertProfile", expertProfileApi.DeleteExpertProfile)           // 删除专家主档
		expertProfileRouter.DELETE("deleteExpertProfileByIds", expertProfileApi.DeleteExpertProfileByIds) // 批量删除专家主档
		expertProfileRouter.PUT("updateExpertProfile", expertProfileApi.UpdateExpertProfile)              // 更新专家主档
	}
	{
		expertProfileRouterWithoutRecord.GET("findExpertProfile", expertProfileApi.FindExpertProfile)       // 根据ID获取专家主档
		expertProfileRouterWithoutRecord.GET("getExpertProfileList", expertProfileApi.GetExpertProfileList) // 分页获取专家主档列表
	}
}
